vaultenv.nix

```diff
@@ -31,8 +31,8 @@ mkDerivation {
       blacklistedSrc = lib.cleanSourceWith {
         src = whitelistedSrc;
         filter = path: type:
-          # Where we're going we don't need no documentation
-          ! lib.hasSuffix ".md" path;
+          # Where we're going we don't need no documentation (nor caches)
+          ! (lib.hasSuffix ".md" path || lib.hasSuffix ".cache" path || lib.hasSuffix "__pycache__" path || lib.hasSuffix ".tar.gz" path);
```

Comment:
We should probably also exclude the `.gitignore`d files:

```nix
      # Remove all things we don't need such as python caches that otherwise mess up the build
      # by causing unnecessary rebuilds due to supposedly changed inputs.
      blacklistedSrc = lib.cleanSourceWith {
        src = whitelistedSrc;
        filter =
          let
            gitignore = builtins.readFile ./.gitignore;
            extraFilter = path: type:
              # Prevent python from messing with the rebuilds
              ! (  lib.hasInfix "/__pycache__/" path
                || lib.hasSuffix "/__pycache__" path
              # Where we're going we don't need no documentation
                || lib.hasSuffix ".md" path
                );
          in
            nix-gitignore.gitignoreFilterPure extraFilter gitignore ./.
          ;
      };
```

What tool created the `.cache` suffix files?

Reply:
> We should probably also exclude the .gitignored files:

done

> What tool created the .cache suffix files?

Some step in the integration tests, but I haven't been able to pinpoint which